apps/src/code-studio/components/pairing/student_selector.jsx

```diff
@@ -4,36 +4,34 @@ import { studentsShape } from './types';
 /**
  * A component for selecting one or more students in a section.
  */
-const StudentSelector = React.createClass({
-  propTypes: {
+export default class StudentSelector extends React.Component {
+  static propTypes = {
     students: studentsShape,
     handleSubmit: PropTypes.func.isRequired
-  },
+  };
 
-  getInitialState() {
-    return {
-      selectedStudentIds: []
-    };
-  },
+  state = {
+    selectedStudentIds: []
+  };
 
-  handleStudentClicked(event) {
-    var selectedStudentIds = this.state.selectedStudentIds;
-    var studentId = +event.target.getAttribute('data-id');
-    var index = selectedStudentIds.indexOf(studentId);
+  handleStudentClicked = (event) => {
+    const selectedStudentIds = [...this.state.selectedStudentIds];
```

Comment:
Does this make a shallow copy?

Reply:
Yep.  The existing code was grabbing a reference to the array inside the component state on line 20, then mutating it directly with `push` or `splice` below before actually calling `setState` on line 30.  In practice I don't think it makes a difference since we were calling `setState` in the end, but I felt better about not making any changes to the component state until the `setState` call.